Меняем Тартарус, чтоб не руинил

TimErmolt opened the pull request:
Хрень срёт в раунд блэкаутами, чужими и метеорами, что его руинит, и на всякий такой раунд возникает туева хуча жалоб в ООС. С задачей сторителлер-то справляется как должен, но за него ради троллинга голосуют мудаки перед каждым раундом. Я не верю в демократию. Пора уже положить этому конец.
Снижение его активности и хардкорности не поможет, потому что либо только чуть-чуть сбавит градус пиздеца, ибо голосовать за него не перестанут, либо уничтожит суть этого характера (он должен вызывать пиздец), что опять же ведёт к удалению. ~~Посему, удаляем сразу.~~
~~Как вариант, конечно, исключить его из голосования и сделать устанавливаемым только педально (педалятор решил поставить -- его проблема, когда раунд скатится в говно), но мне лень. Надо будет -- сделаю.
Билд компилится, так что если сломается на сервере, то что-то там в конфиге мешает. Но не думаю, что должно.~~

**UPD:** Удаления скорее всего не будет, будем менее радикальными способами фиксить Тартарус. Ведём дискуссию в комментариях к ПРу. Новые идеи и проработка уже имеющихся приветствуются.

**UPD2:** Решено оставить Тартарус в билде без изменений, но исключить его из голосования перед раундстартом.

<details>
<summary>Чейнджлог</summary>

```yml
🆑TheUnknownOne
tweak: Личность сторителлера "Тартарус" больше не может быть выбрана в голосовании перед раундстартом.
/🆑
```

</details>

- [x] Pull Request полностью завершен, мне не нужна помощь чтобы его закончить.
- [x] Я внимательно прочитал все свои изменения и багов в них не нашел.
- [x] Я запускал сервер со своими изменениями локально и все протестировал.
- [x] Я ознакомился c [Guide to Contribute](https://github.com/ChaoticOnyx/OnyxBay/blob/dev/docs/contributing.md).

TimErmolt committed:
delete(storyteller): remove tartarus character

It sucks arse and trolls keep voting for it to ruin rounds since it spams blackout, alien infestation, and meteor events. Not needed, thus removing.

TimErmolt commented:
Обобщённое мнение играющих на Хаотике насчёт этого вопроса:
![image](https://user-images.githubusercontent.com/60045066/200020892-b1737fd0-5924-4423-8dd5-686cbf4d8618.png)
(04.11.2022 18:50 по МСК, онлайн ~30 -- участие в опросе 63%)

No-Tips commented:
>игроки хаотика не хотят видеть тартаруса как ст

>игроки хаотика постоянно выбирают тартаруса как ст

Ебанулся?

Zert0X commented:
> Я не верю в демократию. 

Сенат запомнил

TimErmolt commented:
> Ебанулся?

Как видишь, и то и другое заявление верно, исходя из результатов опроса, которые я скинул. Единственный логичный вариант, помимо подкручивания результатов голосования, -- что есть каста троллей, специально его пикающая, чтобы всем насрать, а самим было смешно. Банить за выбор на голосовании -- чёт хуйня, поэтому убираем им возможность руинить игру всем остальным.

FloweYStyle commented:
**Всем привет это я, Флауи. Цветочек по имени Флауи!** Ваш Тартарус каждый раунд спавнит лярв, которые в процессе перерастают в нечестно-сильных ксеноморфов, принуждённо заканчивающих раунд экстренным шаттлом, потому что с ними бороться очень и очень тяжко. 

Это собственно и есть в большинстве своём вся проблема Тартаруса. Ни один раунд с сторителлером Тартарусом не обходится без заспавненной лярвы. Ксеноморфы одни из сильнейших антагов на момент когда я играл, и, судя по последним ПРам - они будут ещё сильнее. Выпиливать Тартарус вполне неплохая идея, поскольку его аналог послабее - Хаотик, вполне справляется с функциями Тартаруса, и, при этом не "срёт" лярвами при любой возможности запустить ивент. 

В моих представлениях "история смерти" должна заключаться не только в спавне ксеноморфов и их последущего развития в непобедимых существ, но и в большом количестве трейторов, генокрадов и прочих прочих антагов, которых люди возможно больше желают видеть, чем надоедливых ксеноморфов.

Sudar0770 commented:
Привет, я Сударь и моё черезмерное эго заставляет меня вставить сюда своё мнение. 

Я согласен с тем, что Тартар срёт однообразием, каждый раунд идёт с начала несколько вирусов, потом пара метеоритных дождей и уже после этого вдруг выясняется, что у квины ксеноморфов уже есть пара охотников. 
Я честно признаюсь, что сам нажимаю при голосовании на Тартар, но не из троллинга, а из-за соблазнения кризисными ситуациями. 
Будет очень жалко если его выпилят, хоть он и скуден на разнообразие, но он не врёт, не обманывает. Он говорит, что нам пиздец и он происходит. 
Мужики, может просто переименуем его в условный "Режим ксеноморфов"? Всё равно все космонавты заранее знают кто их будет трахать, а так хоть при голосовании люди дважды подумают прежде чем его ставить.

Tsurupeta commented:
Клоунада какая-то.

DarkHunter7756 commented:
Привет, я Дарк, и если вам не нравятся ксеноморфы - попробуйте найти к ним тактики, а не ломитесь со всех ног в улей без защиты от фейсхагеров каждый раунд. Спасибо

FloweYStyle commented:
Не припомню тактик против ксеноморфов. Люди опытные и робастные дох сих пор дохнут как мухи от ксеноморфов, а я ничего лучше лимитки в гнездо придумать не могу. Дарк, попробуй поиграть на боевой роли против ксеноморфов. Их скорость, регенерация и оглушение по таклу слишком сильная вещь, чтобы им можно было что-то противопоставить серьёзное. Если именно ксеноморфы почти всегда становятся причиной вызова шаттла, то может быть ксеноморфы действительно сильны, а не игроки стратегии связать против них не могут?
гет рил

Sudar0770 commented:
Значит нужно чтобы при численности ксеноморфов более 3 особей включался карантин и тогда вызова шаттла не будет🤡. 
А если серьёзно, то сейчас решается вопрос не по поводу силы алиенов, они как должны трахать всё не ксеноморфовское, так этим и занимаются. Вопрос в том, почему Тартар срёт ими со 100% вероятностью

TimErmolt commented:
> Значит нужно чтобы при численности ксеноморфов более 3 особей включался карантин и тогда вызова шаттла не будет.

охуеть режим говно так давайте людям даже не дадим возможности от него спастись кроме как выстрелить себе нахуй в рот из ружья

Sudar0770 commented:
> > Значит нужно чтобы при численности ксеноморфов более 3 особей включался карантин и тогда вызова шаттла не будет.
> 
> охуеть режим говно так давайте людям даже не дадим возможности от него спастись кроме как выстрелить себе нахуй в рот из ружья

Чел, я же зарофлил

FloweYStyle commented:
Он пьян, не слушайте его

TimErmolt commented:
> Он пьян, не слушайте его

пососи школьник

TimErmolt commented:
а вообще он прав ну короче суть пра в том что я в рот ебал тартарус и как я сказал его либо выпиливать либо нёрфить но типа бля................. если занёрфить то он смысл потеряет. ну можно сделать особую проверку конечно чтоб он не срал алиумами тупо и будет нормально. но я хз если честно

Zert0X commented:
> если занёрфить то он смысл потеряет

А можно ведь просто сделать ему КД на пару тройку раундов...

TimErmolt commented:
> А можно ведь просто сделать ему КД на пару тройку раундов...

бля ну имхо неплохая идея разве что побольше чем 2-3 раунда. типа ну я хз где-нибудь 10 может. ибо он слишком ядерный чтоб так часто ставиться. почти как ивент армяна.

TimErmolt commented:
а как такой кулдаун реализовать я вообще хз блять

Tsurupeta commented:
мттс 12 часов

TimErmolt commented:
короче я утром просплюсь и распишу весь список возможных альтернатив полному выпилу этого говнища. ждём.

Tsurupeta commented:
мож ты иссуй сделаешь, а не будешь цирк в пре разводить?

TimErmolt commented:
> мож ты иссуй сделаешь, а не будешь цирк в пре разводить?

Иссуи -- это хорошо, но у нас их 2400, и они могут висеть сколько угодно, а шут знает, когда и кто по ним что-нибудь вообще сделает. ПРов в сто раз меньше, и они уже активно предполагают конкретные изменения в билд, что по своей основе делает их рассмотр более срочным и конструктивным. И их всё ещё можно изменить. За цирк извиняюсь, конечно, но я дискуссии в иссуе ждал бы в 10 раз дольше.

Итак, возможные варианты действий по поставленному в ПРе вопросу. Каждый из них как минимум в какой-то степени решит проблему загаживания раундов Тартарусом.

1. Полностью выпилить из билда с концами. Это на данный момент делает пулл реквест.
2. Отключить возможность выбора Тартаруса на голосовании в лобби, но оставить в билде с возможностью педального включения. Педалятор насрёт -- педалятору насрут на лицо всем сервером. По большей части равносильно первому варианту, разве что не так категорично.
3. Понёрфить Тартарус в целом, сделать менее активным. Конечно, вероятно заставит его прекратить руинить раунды, но в то же время опасно приблизится к превращению его в клон Декабря или Хаотика, что не нужно. В этом смысла я не вижу, уж лучше из голосования убрать.
4. Отредактировать список ивентов, вызываемых Тартарусом: в первую очередь, исключить из него алиенов. Ибо именно из-за них возникает большинство жалоб. За место них баффнуть спавн прочих антагов, типа трейторов, генокрадов, может быть, даже революцию или культ, если СБ становится слишком сильной. Один из лучших вариантов, если хочется сделать Тартарус реально хорошим вместо выпила, поскольку решит то, что сейчас он работает по схеме "если прокнет вот этот один ивент (алиены), то станции пизда".
5. Установить кулдаун на возможность выбора Тартаруса в голосовании перед раундом (@Zert0X) -- можно выбрать раз в N раундов. Если выбирается другой сторителлер, то КД не накладывается, очевидно. Сколько раундов кулдауна -- конкретно сказать сложно. Я бы вообще поставил там рандом с не очень большим разбросом, чтобы не считали раунды до появления Тартаруса в кандитатах. Где-нибудь 5-10 включительно.

Мне главное, чтоб мне и другим в раунды перестал билд срать. Поэтому я на самом деле не имею конкретного предпочтения к какой-то из версий (кроме анти-предпочтения к третьей, по описанным там причинам). Так что обсуждаем.

Sudar0770 commented:
Я поддерживаю вариант номер 4 по той причине что:
Я люблю иногда проигрывать в безвыигрышных ситуациях а-ля Самурай идущий на смерть уверенно держащий клинок в руках и Тартар мог бы в теории этому помочь, но очень грустно, что моего самурая каждый раз ебут только ксеноморфы и никто более. 
Я предлагаю снизить шанс спауна ксеноморфов и впилить возможность _только на тартаре_ СБ и прочим не антагнутым профам становиться предателями, еретиками, ксеносами и прочей мразью. На мой взгляд после этого изменения экипажу всё так же не останется надежды на выживание и внесёт ранее невиданный хаос. Тоесть...  Тартар будет давать игрокам пизды в совершенно иной манере что сделает его уникальным и играбельным, а не только пострелякаться с ксеномразями и по итогу пососать. 
Как я вижу примерный раунд на таком тартаре:
Капитан крадёт собственные штаны, ХоС учит СБ магии, один из СБшников эйнахает ХоСа с криками "слава синдикату", а детектив сосёт у смотрителя кровь в техах оправдываясь перед спалившим его клоуном, что он просто на самом деле гей. После половины раунда всё СБ являются культистами-вампирами под нейроимплантами синдиката засваляющим их ходить голышом и кидаются фаерболами в генокрадов-революционеров-коммунистов имеющих базу на судне абдуктора.

NefrNerf commented:
Вы тут все групповую шизу поймали, не удалим Тартар из билда. Предлагаю ввести условия для его активации через голосования, чтобы его можно было выбрать только большинством голосов, а при паритете выбирался какой-то другой.

NefrNerf commented:
Сторителлер выполняет роль "выебать" раунд на отлично, что собственно показывает наличие этого МРа, такой сторителлер нужен. По этому вопрос стоит только в частоте его появлениях и мазохистах которые его ставят.

TimErmolt commented:
> Вы тут все групповую шизу поймали, не удалим Тартар из билда. Предлагаю ввести условия для его активации через голосования, чтобы его можно было выбрать только большинством голосов, а при паритете выбирался какой-то другой.

Как тебе вариант подкрутить ему спавн ивентов, чтобы не спавнились алиены, но с большей частотой спавнились станционные роли и, возможно, что-нибудь ещё? Сударь0770 неплохо в своём комментарии расписал тоже.

TimErmolt commented:
Превратил в драфт, потому что удалять явно не будем. Будем решать, чё с ним делать.

TimErmolt commented:
Кажется, игровое сообщество спустя два месяца наконец осознало, что Тартарус каждый раунд -- говно, а не игра, и перестало за него голосовать. Так что вскоре наверное прикрою ПР, если ничего не поменяется в худшую сторону.

TimErmolt committed:
Revert "delete(storyteller): remove tartarus character"

This reverts commit 7e7fa8a4681b040687c2fd2c58ec59456d3e66b9.

TimErmolt committed:
tweak(storyteller): add can_be_voted_for

Adds `can_be_voted_for` to the storyteller characters.
TRUE by default. When FALSE, this character will not partake in the storyteller vote before roundstart.
Set `can_be_voted_for` to FALSE for Tartarus.

TimErmolt commented:
По заказу от овнеров выбран вариант 2: Тартарус остаётся в билде без изменений, но исключается из раундстартового голосования.
